br.gov.economia.receita.IRPF.yaml

```diff
@@ -0,0 +1,134 @@
+app-id: br.gov.economia.receita.IRPF
+branch: stable
+runtime: org.gnome.Platform
+runtime-version: '41'
+sdk: org.gnome.Sdk
+sdk-extensions:
+  - org.freedesktop.Sdk.Extension.openjdk11
+command: irpf
+finish-args:
+  - --share=network
+  - --socket=x11
+  - --share=ipc
+  - --env=PATH=/app/jre/bin:/app/bin:/usr/bin
+  - --env=JAVA_HOME=/app/jre
+  - --talk-name=org.gtk.vfs.*
+  - --filesystem=xdg-run/gvfsd
+  - --filesystem=~/ProgramasRFB:create
+  # User-defined preferences (such as hiding the welcome window) are stored under: ~/.java/.userPrefs/serpro/ppgd/irpf/prefs.xml
+  - --persist=.java
+  # Log files, tokens and whatnot.
+  - --persist=.receitanet
+  # User-provided form data (such as CPF and CNPJ numbers) are stored under: ~/.rfb/.cacheni.properties
+  - --persist=.rfb
+modules:
+  - name: irpf
+    buildsystem: simple
+    build-commands:
+      - install -Dm755 -t "${FLATPAK_DEST}/bin" "irpf"
+      - install -Dm644 -t "${FLATPAK_DEST}/share/applications" "br.gov.economia.receita.IRPF.desktop"
+      - install -Dm644 "icon-64.png" "${FLATPAK_DEST}/share/icons/hicolor/64x64/apps/br.gov.economia.receita.IRPF.png"
+      - install -Dm644 "icon-128.png" "${FLATPAK_DEST}/share/icons/hicolor/128x128/apps/br.gov.economia.receita.IRPF.png"
+      - install -Dm644 "icon-256.png" "${FLATPAK_DEST}/share/icons/hicolor/256x256/apps/br.gov.economia.receita.IRPF.png"
+      - install -Dm644 -t "${FLATPAK_DEST}/share/metainfo" "br.gov.economia.receita.IRPF.metainfo.xml"
+      - install -Dm644 -t "${FLATPAK_DEST}/share/br.gov.economia.receita.IRPF" "irpf.jar" "pgd-updater.jar"
+      - cp -r --no-preserve=mode help/ lib/ lib-modulos/ "${FLATPAK_DEST}/share/br.gov.economia.receita.IRPF"
+      # Workaround to make the clickable link on the "About" window open with the default web browser.
+      - ln -s "/usr/bin/xdg-open" "${FLATPAK_DEST}/bin/firefox"
+    sources:
+      - type: archive
+        url: https://downloadirpf.receita.fazenda.gov.br/irpf/2022/irpf/arquivos/IRPF2022-1.1.zip
+        sha256: 5ad10513337a56752b12adac0ed03448f4bf4512ef383b5e3713ef3c2bab4322
+      - type: file
+        path: irpf
+      - type: file
+        path: br.gov.economia.receita.IRPF.metainfo.xml
+      - type: file
+        path: br.gov.economia.receita.IRPF.desktop
+      - type: file
+        path: icon-64.png
+      - type: file
+        path: icon-128.png
+      - type: file
+        path: icon-256.png
+  - name: irpf_xml
+    buildsystem: simple
+    build-commands:
+      - install -Dm644 *.xml "${FLATPAK_DEST}/share/br.gov.economia.receita.IRPF/lib/resources"
```

Comment:
As a workaround, you could also symlink this location to a writable path, e.g. a directory under `$XDG_DATA_HOME`.

Reply:
That's a good idea! About the implementation though, I assume I'd have to create and populate this writable directory during runtime, i.e. in the `irpf` script? 🤔

Reply:
Actually, I'm not sure I understand this correctly. I'm thinking I'd have to point `$XDG_DATA_HOME/resources` to `/app/share/br.gov.economia.receita.IRPF/lib/resources` in order for the app to be capable of updating these XML files. But how am I supposed to do that? Because when there is an update, the app tries to overwrite the XML files sitting under `/app/share/br.gov.economia.receita.IRPF/lib/resources` .

Reply:
This tree is read-only.

I wonder if it is not better to get the data as "extra-data".

Reply:
So, something like JDownloader does [currently](https://github.com/flathub/org.jdownloader.JDownloader/blob/7e0c5072d35329d52303940d02c1b3415d6ae72f/org.jdownloader.JDownloader.yaml#L56), I presume?

Everything is "installed" under [`${XDG_DATA_HOME}/jdownloader`](https://github.com/flathub/org.jdownloader.JDownloader/blob/b52b4a27026ca486f53bccb40ccacea38115da38/jd-wrapper.sh#L3) (including the main .jar file), which allows the app to auto-update, but I personally am not a big fan of that approach.

At least for this app, it just silently fails when it can't overwrite these XML files. Besides, I've made a script to help keep them up-to-date with ease.

Reply:
Nevertheless, let me know what approach you guys think it's best.

Reply:
Yeah, that's just a suggestion. Whatever makes your maintenance easier.